src/main/scala/com/microsoft/ml/spark/lightgbm/LightGBMBooster.scala

```diff
@@ -94,7 +130,48 @@ class LightGBMBooster(val model: String) extends Serializable {
         isRowMajor, kind,
         -1, datasetParams, scoredDataLengthLongPtr, scoredDataOutPtr),
       "Booster Predict")
-    predToArray(classification, scoredDataOutPtr, kind)
+    predScoreToArray(classification, scoredDataOutPtr, kind)
+  }
+
+  protected def predictLeafForCSR(sparseVector: SparseVector): Array[Double] = {
+    val numCols = sparseVector.size
+
+    val datasetParams = "max_bin=255 is_pre_partition=True"
+    val dataInt32bitType = lightgbmlibConstants.C_API_DTYPE_INT32
+    val data64bitType = lightgbmlibConstants.C_API_DTYPE_FLOAT64
+
+    ensureLeafIndexDataCreated()
+
+    LightGBMUtils.validate(
+      lightgbmlib.LGBM_BoosterPredictForCSRSingle(
+        sparseVector.indices, sparseVector.values,
+        sparseVector.numNonzeros,
+        boosterPtr, dataInt32bitType, data64bitType, 2, numCols,
+        lightgbmlibConstants.C_API_PREDICT_LEAF_INDEX, -1, datasetParams,
+        leafIndexDataLengthLongPtr, leafIndexDataOutPtr), "Booster Predict")
```

Comment:
for better debugging change error string to:
"Booster Predict" -> "Booster Predict Leaf"
(this is the string displayed in LightGBMUtils.validate when the called method returns error)